src/TaskQueue.js

```diff
@@ -0,0 +1,77 @@
+const iterate = require('taskcluster-lib-iterate');
+const assert = require('assert');
+const _ = require('lodash');
+class TaskQueue {
+  constructor(cfg, queue) {
+    assert(cfg.worker.workerId, 'Worker ID is required');
+    assert(cfg.worker.workerType, 'Worker type is required');
+    assert(cfg.worker.workerGroup, 'Worker group is required');
+    assert(cfg.worker.provisionerId, 'Provisioner ID is required');
+    assert(queue, 'Instance of taskcluster queue is required');
+    this.queue = queue;
+    this.workerType = cfg.worker.workerType;
+    this.provisionerId = cfg.worker.provisionerId;
+    this.workerGroup = cfg.worker.workerGroup;
+    this.workerId = cfg.worker.workerId;
+  }
+  async startWorker() {
+
+    while (true) {
+      var res = await this.claimTasks();
+    }
+  }
+
+  async claimTasks() {
+    var capacity = 1;
+    let result = await this.queue.claimWork(this.provisionerId, this.workerType, {
+      tasks: capacity,
+      workerGroup: this.workerGroup,
+      workerId: this.workerId,
+    });
+    let stat = '';
+    if (_.isEmpty(result.tasks.task.payload)) {
+      if (result.tasks.task.workerType === 'succeed') {
+        stat = await this.successResolver(result);
+      } else if (result.tasks.task.workerType === 'fail') {
+        stat = await this.failureResolver(result);
+      }
+    } else {
+      stat = await this.malformedPayload(result);
+    }
+    return stat;
+  }
+
+  async successResolver(result) {
```

Comment:
I think we can get rid of these functions, they are too small. You can just have these statements in your `claimTasks`

Too many of functions that are too small make code hard to read and understand. Also, functions are expensive in JavaScript, and too many functions can affect performance and cause memory leaks, especially in the browser. 

Typically, if your function just returns another function or result of another function without adding anything or processing anything - it's definitely too small. In this case, you basically have `return await this.queue.reportCompleted(result.tasks.status.taskId, result.tasks.runId);` <---- no need to wrap this in another function.